### PR TITLE
Added GPA field to backend

### LINKED ISSUE #8: [Status]: Re-implement the status field
The status field will be used in our application dashboard to organize the applicants.

Core functionality - We made this a core functionality because without the status field, we wouldn't be able to place the students into different status columns in our dashboard.
**Estimated effort**: 1-2 hours. The feature that we implemented in the last homework (status) would need to be reused. This would involve using the main branch of the status feature repo. 
**Dependencies**: None
**Acceptance criteria**: The status field should be implemented as it was in Homework 2.

### LINKED ISSUE #9: [GPA]: Add GPA range column to Document backend
core functionality
The GPA range column needs to be added to the backend so that the document model /database have the updated fields.
We set this to core functionality because without a GPA range column, users would be unable to sort by range.
**Estimated effort**: 1 day. We would need to create the GPA column in Document.java, update the getDoc/update in DAO with the column, add a database migration, and update the DTO interface similarly (and somewhat identically) to how the backend was changed for status field. 
**Dependencies**: None
**Acceptance criteria**: Document model, DAO/DTO and DB are updated with GPA field and provides connection to middleware.

### PR DESCRIPTION
**Pull request by ndnrlg**
Resolves #9, stores the GPA in the backend so that it is persistent. This gives middleware access to GPA_C column in database.

image (1)

### CODE REVIEW

**Comment by pranavaddepalli** [comment]
@ndnrlg conflicts

**Review by pranavaddepalli — CHANGES_REQUESTED** [review]
fix conflicts

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Document.java [hunk 1]
```diff
@@ -47,6 +47,12 @@ public class Document implements Loggable {
     @Column(name = "DOC_TITLE_C", nullable = false, length = 100)
     private String title;
     
+    /**
+     * GPA range.
+     */
+    @Column(name = "DOC_GPA_C", nullable = false, length = 20)
+    private String gpa = "3.7-4.0";
```
Comment (pranavaddepalli) [inline docs-core/src/main/java/com/sismics/docs/core/model/jpa/Document.java:60]: not sure why set to this

File: docs-core/src/main/java/com/sismics/docs/core/service/InboxService.java [hunk 2]
```diff
@@ -229,6 +229,7 @@ private void importMessage(Message message, Map<String, String> tags) throws Exc
         document.setFormat("EML");
         document.setSource("Inbox");
         document.setLanguage(ConfigUtil.getConfigStringValue(ConfigType.DEFAULT_LANGUAGE));
+        document.setGPA("3.7-4.0");
```
Comment (pranavaddepalli) [inline docs-core/src/main/java/com/sismics/docs/core/service/InboxService.java]: same here

File: docs-core/src/test/java/com/sismics/docs/core/util/TestFileUtil.java [hunk 3]
```diff
@@ -94,6 +94,7 @@ public void convertToPdfTest() throws Exception {
             documentDto.setCoverage("France");
             documentDto.setRights("Public Domain");
             documentDto.setLanguage("en");
+            documentDto.setGPA("3.7-4.0");
```
Comment (pranavaddepalli) [inline docs-core/src/test/java/com/sismics/docs/core/util/TestFileUtil.java]: this one might be fine because it is part of the tests

**Review by pranavaddepalli — CHANGES_REQUESTED** [review]
See comments

File: docs-core/src/main/java/com/sismics/docs/core/model/jpa/Document.java (hunk 1 above)
Comment (ndnrlg) [reply docs-core/src/main/java/com/sismics/docs/core/model/jpa/Document.java:60]: This had been fixed through (uncommited) merged conflicts. The dev branch is now updated so gpa's default value is "N/A".

**Pull request merged**